src/utils/miscellaneous.ts

```diff
@@ -217,15 +218,14 @@ export interface TimeoutParams<T> {
   timeoutErrorMsg?: string
 }
 
-export function timeout<T>(params: TimeoutParams<T>): Promise<T> {
+export function timeout(params: TimeoutParams<undefined>): Promise<never> // never means function throws
+export function timeout<T>(params: TimeoutParams<T extends undefined ? never : T>): Promise<T>
+export async function timeout<T>(params: TimeoutParams<T>): Promise<T | never> {
```

Comment:
this should be renamed to promisedTimeout or sth, to indiciate it is `await`able in nature or actually i dont care

Reply:
hm, I also don't really care